HLTrigger/Timer/interface/FastTimerService.h

```diff
@@ -214,6 +216,10 @@ class FastTimerService {
   void preEventReadFromSource(edm::StreamContext const&, edm::ModuleCallingContext const&);
   void postEventReadFromSource(edm::StreamContext const&, edm::ModuleCallingContext const&);
 
+  // inherited from TBB task_scheduler_observer
+  void on_scheduler_entry(bool worker) final;
+  void on_scheduler_exit(bool worker) final;
+
```

Comment:
@Dr15Jones do you think it might be useful to monitor these also in other services, like the Tracer ?

Reply:
We already monitor it in the code that does the crash logging (in order to capture which modules are in which threads). I had not contemplated where else it might be useful.

Reply:
OK. I've added it to the `FastTimerService` to track the resources used in a new thread before it starts working on some module.

Reply:
Be forewarned, the framework turns TBB on right up front during the initialization phase which is well before the framework explicitly starts to create TBB tasks. Therefore you should see lots of 'dead' startup time.

Reply:
[NOTE: it is possible that TBB does 'lazy' thread creation and therefore waits until the first TBB task is requested before it starts creating the threads. I think that better explains what I see in the debugger so you might not have much 'dead' startup time.]